### PR TITLE
Add note about versioning

### PR DESCRIPTION
**Pull request by riyazdf**
Adding a note to the docs on ensuring that the notary client and server/signer are the same version, also fixed a small typo

Signed-off-by: Riyaz Faizullabhoy <riyaz.faizullabhoy@docker.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,6 +42,7 @@ Any security vulnerabilities can be reported to security@docker.com.
 # Getting started with the Notary CLI
 
 Please get the Notary Client CLI binary from [the official releases page](https://github.com/docker/notary/releases) or you can [build one yourself](#building-notary).
+The version of Notary server and signer should match the Notary CLI to ensure feature compatibility, and all official releases are associated with GitHub tags.
```
Comment (cyli) [inline README.md]: Should we say that the server version should be >= the client version?  Newer versions of the server should also work with older versions of clients.

Comment (riyazdf) [inline README.md]: Yeah, definitely!  Updated.

**Comment by endophage** [comment]
Would be good to include an example: `i.e. if you're using notary 0.2, ensure you are using a server and signer tagged with an equal or higher version than 0.2`

**Comment by cyli** [comment]
LGTM!  Thank you for adding this!!

**Comment by endophage** [comment]
LGTM

**Pull request merged**